app.py

```diff
@@ -119,13 +133,33 @@ def post(self):
         self.finish(json.dumps(finish_ret))
 
 
+def get_user_img_prefix(self):
+    cookie = self.get_cookie(app_cookie)
+    user_id = cookie if cookie else ""
+    return temp_img_prefix + user_id + "-"
+
+
+def valid_user_img(self, img):
+    default_img = not img.startswith(static_img_path + temp_img_prefix)
+    user_img = img.startswith(static_img_path + get_user_img_prefix(self))
+    current_user_img = user_img if self.get_cookie(app_cookie) else False
+    return default_img or current_user_img
+
+
+def get_image_captions(self):
+    return collections.OrderedDict(
+        (k, v) for k, v in image_captions.items() if valid_user_img(self, k)
+    )
+
+
 def run_ml_queued(img_path, ret_queue):
     caption = run_ml(img_path)
     ret_queue.put((img_path, caption))
 
 
 def valid_file_ext(filename):
-    return '.' in filename and filename.split('.', 1)[1].lower() in VALID_EXT
+    filename, file_extension = os.path.splitext(filename)
+    return file_extension.lower() in VALID_EXT
```

Comment:
add logging here? would have helped with your other bug.

Reply:
I had a `print` while testing, but it didn't seem worth adding a `log` line. What string would you recommend to log here? I don't want to cluster the logs too much.

Reply:
log when it's not valid? i'm assuming that would have caught the `.` in filename bug ?

```
if file_extension.lower() not in VALID_EXT
  log something
return
```

Reply:
ah, gotcha